specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2024-03-01-preview/AlertProcessingRules.json

```diff
@@ -0,0 +1,900 @@
+      "required": [
+        "actionGroupIds"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/Action"
+        }
+      ],
+      "properties": {
+        "actionGroupIds": {
+          "description": "List of action group Ids to add to alert processing rule.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "RemoveAllActionGroups": {
+      "x-ms-discriminator-value": "RemoveAllActionGroups",
+      "type": "object",
+      "description": "Indicates if all action groups should be removed.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Action"
+        }
+      ]
+    },
+    "CorrelateAlerts": {
+      "x-ms-discriminator-value": "CorrelateAlerts",
+      "type": "object",
+      "description": "Add logic for alerts correlation.",
+      "required": [
+        "correlateBy",
+        "correlationInterval",
+        "priority"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/Action"
+        }
+      ],
+      "properties": {
+        "correlateBy": {
+          "type": "array",
+          "description": "The list of conditions for the alerts correlations.",
+          "items": {
+            "$ref": "#/definitions/CorrelateBy"
+          },
+          "x-ms-identifiers": []
+        },
+        "correlationInterval": {
+          "type": "string",
+          "description": "The required duration (in ISO8601 format) for the alerts correlation."
+        },
+        "priority": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The priority of this correlation."
+        },
+        "notificationsForCorrelatedAlerts": {
+          "type": "string",
+          "default": "SuppressAlways",
+          "description": "Indicates how to handle child alerts notifications.",
+          "enum": [
+            "NotifyAlways",
+            "SuppressAlways"
+          ],
+          "x-ms-enum": {
+            "name": "NotificationsForCorrelatedAlerts",
+            "modelAsString": true
+          }
+        },
+        "correlationUpdates": {
+          "$ref": "#/definitions/CorrelationUpdates"
+        }
+      }
+    },
+    "CorrelateBy": {
+      "type": "object",
+      "description": "The logic for the correlation.",
+      "properties": {
+        "field": {
+          "type": "string",
+          "description": "The JPath of the property that the alerts should be correlated by."
+        }
+      }
+    },
+    "CorrelationUpdates": {
+      "type": "object",
+      "description": "Defines the updates to be applied for correlation.",
+      "properties": {
+        "updateType": {
+          "type": "string",
+          "description": "The type of update that needs to be performed."
+        },
+        "updateInterval": {
+          "type": "string",
+          "description": "The update interval in ISO8601 format."
+        },
```

Comment:
For sake of API users can you add further info in the description about how this differs from correlationInterval?